core/templates/components/question-directives/questions-list/questions-list.component.ts

```diff
@@ -273,51 +273,69 @@ export class QuestionsListComponent implements OnInit, OnDestroy {
     this.windowRef.nativeWindow.location.hash = this.questionId;
   }
 
-  deleteQuestionFromSkill(
-      questionId: string, skillDescription: string): void {
+  async isQuestionDeletionAllowed(): Promise<boolean> {
```

Comment:
Are you deleting the question from the database, or unlinking it from the skill (but the question still exists)?

These are two different things, and the naming should reflect that.

Reply:
Updated the naming in every place to reflect `remove` instead of `delete`. Please take a look.